packages/components/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,181 @@
+  }
+`;
+
+export const CheckIndicator = (): React.ReactElement => {
+  return (
+    <StyledCheckboxIndicator as={RadixIndicator} data-testid="checkbox-check">
+      <Icon
+        color="colorBackground"
+        decorative={true}
+        icon="CheckIcon"
+        size="sizeIcon30"
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export const IndeterminateIndicator = (): React.ReactElement => {
+  return (
+    <StyledCheckboxIndicator
+      as={RadixIndicator}
+      data-testid="checkbox-indeterminate"
+    >
+      <Icon
+        color="colorBackground"
+        decorative={true}
+        icon="MinusSmallIcon"
+        size="sizeIcon30"
+      />
+    </StyledCheckboxIndicator>
+  );
+};
+
+export interface CheckboxRootProps extends RadixCheckboxProps {
+  wrapped?: boolean;
+  error?: boolean;
+  checkboxId: string;
+  CheckboxIcon?: React.FunctionComponent;
+  children: NonNullable<React.ReactNode>;
+}
+
+const getCheckboxStyles = (
+  wrapped: boolean
+): {
+  backgroundColor: SystemProp<keyof Theme["colors"], Theme>;
+  padding: SystemProp<keyof Theme["space"], Theme>;
+} => {
+  if (wrapped) {
+    return {
+      backgroundColor: "colorBackgroundWeak",
+      padding: "space40",
+    };
+  }
+  return {
+    backgroundColor: "colorBackground",
+    padding: "space0",
+  };
+};
+
+const StyledCheckboxRoot = styled.input`
+  width: 16px;
+  height: 16px;
+  border-radius: 4px;
+  margin-right: ${theme.space.space30};
+  padding: 4px 0 0;
+
+  &:focus {
+    box-shadow: 0 1px 2px rgba(0, 0, 0, 0.05), 0 0 0 2px ${
+      theme.colors.colorBackground
+    }, 0 0 0 4px ${theme.colors.colorBackgroundPrimary};
+    outline: none;
+  },
+
+  &[data-state=unchecked] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${(props: { error: boolean }) =>
+      props.error
+        ? theme.colors.colorBackgroundDestructive
+        : theme.colors.colorBorder};
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBorder};
+    },
+  },
+  
+  &[data-state=checked], &[data-state=indeterminate] {
+    background-color: ${theme.colors.colorBackgroundPrimary};
+    border: 1px solid ${theme.colors.colorBackgroundPrimary};
+
+    &[data-disabled=true] {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    }
+
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundPrimaryStrong};
+      border: 1px solid ${theme.colors.colorBackgroundPrimaryStrong};
+    },
+  },
+  
+  &[data-disabled=true] {
+    background-color: ${theme.colors.colorBackground};
+    border: 1px solid ${theme.colors.colorBackgroundStrong};  
+  
+    &:hover {
+      background-color: ${theme.colors.colorBackgroundStrong};
+      border: 1px solid ${theme.colors.colorBackgroundStrong};
+    },
+  },
+`;
+
+export const CheckboxRoot = React.forwardRef<
+  HTMLButtonElement,
```

Comment:
Are you sure that it is correct? A checkbox implements the `HTMLInputElement`.

Reply:
Radix's checkbox uses a button under the hood so it's important to leave it as is